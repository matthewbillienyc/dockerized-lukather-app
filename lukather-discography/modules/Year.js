import React from 'react'
import 'whatwg-fetch'

export default React.createClass({
  getInitialState(){
    return { yearData: { year: '', albums: [] } }
  },
  
  getYear(){
    let self = this
    fetch(`http://localhost:3000/v1/years/${this.props.params.year}`)
      .then(function(response){
        return response.json()
      })
      .then(function(json){
        self.setState({ yearData: json.year })
      })
  },
  
  render() {
    this.getYear()
    let albumsList = this.state.yearData.albums.map(function(album){
      return (<li key={album.title + album.artist}>{album.title} - {album.artist}</li>)
    })
    return (
      <div>
        <h2>{this.props.params.year}</h2>
        <ul className='album-list'>
          {albumsList}
        </ul>
      </div>
    )
  }
})
