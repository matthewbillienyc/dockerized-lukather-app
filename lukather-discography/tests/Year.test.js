import React from 'react'
import Year from '../modules/Year'
import renderer from 'react-test-renderer'

test('Year creates list items for each album', () => {
  let props = {
    year: '1977',
    albums: [
      {
        title: 'Rock On',
        artist: 'MJ'
      },
      {
        title: 'Album 2',
        artist: 'Artist 2'
      }
    ]
  }
  
  const component = renderer.create(
    <Year currentYear={props}/>
  )

  let tree = component.toJSON()
  
  expect(tree).toMatchSnapshot()
})
