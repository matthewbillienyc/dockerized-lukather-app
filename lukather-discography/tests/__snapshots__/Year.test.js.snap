exports[`test Year creates list items for each album 1`] = `
<div>
  <ul
    className="album-list">
    <li>
      Rock On
       - 
      MJ
    </li>
    <li>
      Album 2
       - 
      Artist 2
    </li>
  </ul>
</div>
`;
